### DISCUSSION TITLE
Generic Method Pattern

### DISCUSSION THREAD

**Discussion by waldo1001** [microsoft/alguidelines#41] [BC Patterns]
Any feedback on this pattern is always appreciated: questions, remarks, doubts, advantages, disadvantages, ... .

You can find the pattern here: https://alguidelines.dev/docs/patterns/generic-method-pattern/

**Comment by m3koenig** [comment]
Hi :)
I did not like the way the procedure is called.
Like the example: `WLDBlockCustomerMeth.BlockCustomer(Rec, HideDialog);` 
In my opinion its hard to read. Especially when the method names get longer.
What did you think about overloading the "Run" method of a Codeunit?
![image](https://github.com/user-attachments/assets/f1b58eb0-00c4-408e-ae1a-50f78ae1cfb5)
